common/src/bitset.rs

```diff
@@ -196,6 +196,30 @@ impl BitSet {
         Ok(())
     }
 
+    pub fn deserialize(mut reader: impl Read) -> io::Result<Self> {
+        let mut max_buf = vec![0; 4];
```

Comment:
i have a different suggestion below, but you can avoid a heap allocation by using an array on the stack for these small buffers:
```rust
let mut max_buf = [0; 4];
reader.read_exact(&mut max_buf)?;
```